### ISSUE TITLE
Sampling rate modifications

### ISSUE THREAD

**Issue by pascalfiv** [ARBML/klaam#5]
Hello @zaidalyafeai 

For our bachelor thesis a friend and me started working on dialect classifcation a while ago, now we came across your repo and your working with the same corpus as we did. We want to investigate how the length of the provided samples is influencing the trained classifier when using wav2vec-xlsr as the base-model. 

After some investigation of your code, we were wondering why you just read the first 20 seconds of each file. Is this not somewhat contraproductive? As we are losing a lot of training data trough that?

```python
 def speech_file_to_array_fn(batch):
        start = 0 
        stop = 20 
        srate = 16_000
        speech_array, sampling_rate = sf.read(batch["file"], start = start * srate , stop = stop * srate)
        batch["speech"] = librosa.resample(np.asarray(speech_array), sampling_rate, srate)
        batch["sampling_rate"] = srate
        batch["parent"] = batch["label"]
        return batch
```

**Did you preprocess your data cutted into smaller pieces so each is max. 20seconds long?** Or is it possible to read the whole files in so and generate our batches according to the length of each file. As the whole thing is not quite straight forward to implement.

**Comment by zaidalyafeai** [comment]
The main problem we ran into was the variation of length of files ranging from a few seconds into a few minutes. Unfortunately, fitting minutes of data inside a 16 GB GPU (training was all in google colab) was not possible, so we had to discard some of the data. It is always possible to test with different lengths and observe if that improves accuracy. Another idea is to chop each file into equal lengths  and include the whole data in training but that might cause bias towards speakers with larger files.

**Comment by pascalfiv** [comment]
yep makes sense. 
So you did train the whole thing on google colab?  In the end many of the files aren't much longer than 20 seconds so it isn't much of a problem anyway.

Yeah we did also evaluate your uploaded model on huggingface and we ahchieve same result as you did. 
Great work anyways! :)

**Comment by zaidalyafeai** [comment]
Yes, we trained most of the models on google colab.

**Comment by zaidalyafeai** [comment]
closing for now, feel free to reopen.

**Comment by pascalfiv** [comment]
Hi @zaidalyafeai 

We are trying to retrain your model to achieve the same result but we are wondering, what train-settings you actually took, I don't think that it's the same as described down here, we have sadly only one GPU with 12 Gigs of ram but we want to achieve your result with an own train-run that's why we are asking:
```bash
python run_classifier.py \
    --model_name_or_path="facebook/wav2vec2-large-xlsr-53" \
    --output_dir=/path/to/output \
    --cache_dir=/path/to/cache/ \
    --freeze_feature_extractor \
    --num_train_epochs="50" \
    --per_device_train_batch_size="32" \
    --preprocessing_num_workers="1" \
    --learning_rate="3e-5" \
    --warmup_steps="20" \
    --evaluation_strategy="steps"\
    --save_steps="100" \
    --eval_steps="100" \
    --save_total_limit="1" \
    --logging_steps="100" \
    --do_eval \
    --do_train \
```

**Comment by zaidalyafeai** [comment]
On colab 16 batch size works.

```
python run_classifier.py \
    --model_name_or_path="facebook/wav2vec2-large-xlsr-53" \
    --output_dir=/content/drive/MyDrive/Speech/dialects-classification \
    --cache_dir=/content/data_dir/ \
    --freeze_feature_extractor \
    --num_train_epochs="5" \
    --per_device_train_batch_size="16" \
    --preprocessing_num_workers="1" \
    --learning_rate="3e-5" \
    --warmup_steps="20" \
    --evaluation_strategy="steps"\
    --save_steps="1000" \
    --eval_steps="1000" \
    --save_total_limit="1" \
    --logging_steps="1000" \
    --do_eval \
    --do_train \
```

**Comment by ishatariq78** [comment]
Hi @zaidalyafeai i am trying to train text to speech model for arabic common voices. can you help me with data preprocessing as training is giving error. like i have cips and train.tsv file. what should i do next.
